encoding/json/encoding_test.go

```diff
@@ -3024,6 +3024,81 @@ func TestDecodeFixedPoints(t *testing.T) {
 	})
 }
 
+func TestDecodeDeprecatedTypes(t *testing.T) {
+
+	t.Parallel()
+
+	t.Run("with static reference type", func(t *testing.T) {
+
+		t.Parallel()
+
+		testDecode(
+			t,
+			// language=json
+			`
+              {
+                "type": "Type",
+                "value": {
+                  "staticType": {
+                    "kind": "Reference",
+                    "type": {
+                      "kind": "Int"
+                    },
+                    "authorized": true 
+                  }
+                }
+              }
+            `,
+			cadence.TypeValue{
+				StaticType: &cadence.DeprecatedReferenceType{
+					Authorized: true,
+					Type:       cadence.IntType,
+				},
+			},
+			WithBackwardsCompatability(),
```

Comment:
Also test without backward compatibility, same below

Reply:
added additional tests. The reference type will check for error, while the restricted one will check for panic 👍